client/src/AddVideoForm.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from "react";
+import { Modal, Button } from "react-bootstrap";
+
+function AddVideoForm({ onAddVideo, fetchVideos }) {
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+  const [genre, setGenre] = useState("");
+  const [filledForm, setFilledForm] = useState(null);
+  const [errorPopUp, setErrorPopUp] = useState(false);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (!genre || !title || !url) {
+      setErrorPopUp(true);
+      return;
+    }
+
+    fetch("http://localhost:5000/videos/addnew", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        genre: genre,
+        title: title,
+        url: url,
+      }),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data.message);
+        setFilledForm(data.message);
+        fetchVideos();
+        setTitle("");
+        setUrl("");
+        setGenre("");
+      })
+      .catch((error) => {
+        console.log(error);
```

Comment:
Nice remembering to handle errors!

Reply:
Yeeesss....I learned the way way Sam....because my project kept crashing annnnd ... I wanted my clients/end users aware of whats up!!😁